Implement sorting by stars in song library

Dansla116 opened the pull request:
Backend:
- Updated `SongEntry` and sorting logic in **YARG.Core**
- Integrated new sort mode into the `SongCache` system
- Refactored `SongEntry` and `SongCache` integration for cleaner star sorting
- Ensured star query logic resides in **ScoreDatabase.cs**
- Triggered music library refresh automatically when star ratings change

Frontend:
- Added new menu UI for selecting star sorting option
- Updated assets and `en-US.json` localization file
- Adjusted UI assets for consistent visual alignment
- Grouped songs contextually by instrument and difficulty
- Added secondary sort by intensity, tertiary sort by name

Other:
- Ensured compatibility with existing sorting workflows

Notes:
- I wasn’t sure where to best place the interfaces that connect `QueryPlayerSongBestStars()` results from **ScoreDatabase** to **YARG.Core/SongEntry**. Would appreciate any feedback or suggestions on this workflow.
- Songs are primarily grouped and sorted by number of stars descending, then secondarily by difficulty of the instrument you're playing ascending, and tertiarily by song name ascending.  If memory serves, that's how Rock Band 4 did it.  If my memory is wrong or we'd prefer another way, I'm open to suggestions.
- **Performance:** The music library currently refreshes whenever instrument, difficulty, or number of stars on any **one song** changes (typically after every song).  On my system, this adds ~1 second per 3000 songs.  Let me know if anyone thinks this needs further optimization.

sonicfind commented:
One thing I will consider is potentially having the StarAmount enum as a variable of SongEntry; however, you can not involve it in the cache in any way nor in the core-side song sorting. It's purpose would simply be to connect the database star amounts from the main repo to their accompanying entries after the fact.

Dansla116 commented:
Removed all prior changes. Sorting by stars no longer touches `SongEntry` or `SongCache`, and the interfaces for communicating between YARG.Core and the main YARG module are no longer needed and have been removed.  The only remaining change in this PR is the expansion of the StarAmount enum, which now includes the helper methods that were previously part of the (now deleted) StarSortHelper.cs from the YARG supermodule.

sonicfind commented on YARG.Core/Game/StarAmount.cs:
```diff
@@ -30,9 +33,41 @@ public static int GetStarCount(this StarAmount starAmount)
+            };
+        }
+
+        public static string GetDisplayName(this StarAmount stars)
+        {
+            return stars switch
+            {
+                StarAmount.StarGold => "Gold Stars",
+                StarAmount.Star5    => "5 Stars",
+                StarAmount.Star4    => "4 Stars",
+                StarAmount.Star3    => "3 Stars",
+                StarAmount.Star2    => "2 Stars",
+                StarAmount.Star1    => "1 Star",
+                StarAmount.None     => "UNPLAYED SONGS",
+                StarAmount.NoPart   => "NO PART",
+                _                   => "Other"
+            };
+        }
```
Additionally, with some changes to the main repo PR, this can just become an array localized to `SongContainer`.

Dansla116 commented:
Pushed a change addressing feedback - changed StarAmount from `byte` to `sbyte`, assign -1 to NoPart and removed the "_ => -2" fallback for the fallback.
